w3w-autosuggest/public/js/w3w-autosuggest-public.js

```diff
@@ -30,178 +30,213 @@
-        W3W_AUTOSUGGEST_SETTINGS.clip_to_circle_radius,
-      ].join(',')
-      w3wComponent.setAttribute('clip_to_circle', circle)
-    }
-
-    return w3wComponent
-  }
-
-  if (W3W_AUTOSUGGEST_SETTINGS.woocommerce_enabled) {
-    const billingCountry = $('#billing_country')
-    const shippingCountry = $('#shipping_country')
-    const billingTarget = document.querySelector('#w3w-billing')
-    const shippingTarget = document.querySelector('#w3w-shipping')
-
-    if (billingCountry && billingTarget) {
-      const billingTargetSibling = billingTarget.nextSibling
-      const billingTargetParent = billingTarget.parentElement
-      const billingW3wComponent = createAutosuggestComponent(billingTarget, billingTargetParent, billingTargetSibling)
-      billingW3wComponent.setAttribute('name', 'billing_w3w')
-      
-      if (W3W_AUTOSUGGEST_SETTINGS.save_nearest_place) {
-        billingW3wComponent.addEventListener('selected_suggestion', function(e) {
-          const nearestPlace = e.detail.suggestion.nearestPlace
-          $('#billing_nearest_place').attr('value', nearestPlace)
-        })
+      if (W3W_AUTOSUGGEST_SETTINGS.enable_clip_to_bounding_box) {
+        const bounding_box = [
+          W3W_AUTOSUGGEST_SETTINGS.clip_to_bounding_box_ne_lat,
+          W3W_AUTOSUGGEST_SETTINGS.clip_to_bounding_box_ne_lng,
+          W3W_AUTOSUGGEST_SETTINGS.clip_to_bounding_box_sw_lat,
+          W3W_AUTOSUGGEST_SETTINGS.clip_to_bounding_box_sw_lng,
+        ].join(',')
+        w3wComponent.setAttribute('clip_to_bounding_box', bounding_box)
       }
-
-      if (W3W_AUTOSUGGEST_SETTINGS.return_coordinates) {
-        billingW3wComponent.addEventListener('coordinates_changed', function(e) {
-          const coordinates = e.detail.coordinates
-          $('#billing_w3w_lat').attr('value', coordinates.lat)
-          $('#billing_w3w_lng').attr('value', coordinates.lng)
-        })
+      if (W3W_AUTOSUGGEST_SETTINGS.enable_clip_to_circle) {
+        const circle = [
+          W3W_AUTOSUGGEST_SETTINGS.clip_to_circle_lat,
+          W3W_AUTOSUGGEST_SETTINGS.clip_to_circle_lng,
+          W3W_AUTOSUGGEST_SETTINGS.clip_to_circle_radius,
+        ].join(',')
+        w3wComponent.setAttribute('clip_to_circle', circle)
       }
-      
-      billingW3wComponent.appendChild(billingTarget)
-      billingTargetParent.insertBefore(billingW3wComponent, billingTargetSibling)
-
-      if (
-        !W3W_AUTOSUGGEST_SETTINGS.enable_clip_to_country &&
-        !W3W_AUTOSUGGEST_SETTINGS.enable_clip_to_circle &&
-        !W3W_AUTOSUGGEST_SETTINGS.enable_clip_to_bounding_box &&
-        !W3W_AUTOSUGGEST_SETTINGS.enable_clip_to_polygon
-      ) {
-        console.log('inside')
-        billingCountry.on('change', function(e) {
-          $('#w3w-billing').closest('what3words-autosuggest').attr('clip_to_country', e.target.value)
-        })
-        billingCountry.trigger('change')
+
+      if (isPartialWooCommerce) {
```

Comment:
This is the new scenario. 
If there's a better way to determine if the field being used is billing- or shipping-related, please let me know. This version seems a bit fragile.

Reply:
This could also be added to the PHP `/w3w-autosuggest/admin/class-w3w-autosuggest-admin.php` and adding to the `$settings` that feeds `W3W_AUTOSUGGEST_SETTINGS` in the JS file as a boolean value. This might be a bit more reliable. https://woocommerce.com/document/query-whether-woocommerce-is-activated/